packages/runtime/src/runtime.ts

```diff
@@ -1036,4 +1038,70 @@ export class Runtime {
 			throw new RuntimeError(RUNTIME_ERRORS.GENERAL.INVALID_SECRET_KEY);
 		}
 	}
+
+	/**
+	 * Verifies multi-signature and throws an error if signatures are not valid
+	 * @param signedTransaction signedTransaction object
+	 */
+	validateMultisignature(signedTransaction: algosdk.SignedTransaction) {
+		this.verifyMultisig(signedTransaction);
+		if (!this.verifyMultisig(signedTransaction)) {
+			throw new RuntimeError(RUNTIME_ERRORS.GENERAL.INVALID_MULTISIG);
+		}
+	}
+
+	/**
+	 * Verify multi-signature
+	 * @param signedTxn signedTransaction object
+	 */
+	/* eslint-disable sonarjs/cognitive-complexity */
+	verifyMultisig(signedTxn: SignedTransaction): boolean {
+		if (signedTxn.msig === undefined) {
+			return false;
+		} else {
+			const version = signedTxn.msig.v;
+			const threshold = signedTxn.msig.thr;
+			const subsigs = signedTxn.msig.subsig;
+			const addrs = subsigs.map((subsig) => algosdk.encodeAddress(subsig.pk));
+			if (signedTxn.msig.subsig.length < threshold) {
+				return false;
+			}
+			let multiSigaddr;
```

Comment:
variables should name follow cameo rule
```suggestion
			let multisigAddr;
```